components/writers/src/main/java/org/datacleaner/beans/writers/UpdateDataHubAnalyzer.java

```diff
@@ -0,0 +1,572 @@
+import org.apache.metamodel.util.Resource;
+import org.datacleaner.api.Analyzer;
+import org.datacleaner.api.Categorized;
+import org.datacleaner.api.ColumnProperty;
+import org.datacleaner.api.ComponentContext;
+import org.datacleaner.api.Concurrent;
+import org.datacleaner.api.Configured;
+import org.datacleaner.api.Description;
+import org.datacleaner.api.ExecutionLogMessage;
+import org.datacleaner.api.FileProperty;
+import org.datacleaner.api.FileProperty.FileAccessMode;
+import org.datacleaner.api.HasLabelAdvice;
+import org.datacleaner.api.Initialize;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.api.InputRow;
+import org.datacleaner.api.MappedProperty;
+import org.datacleaner.api.Provided;
+import org.datacleaner.api.SchemaProperty;
+import org.datacleaner.api.TableProperty;
+import org.datacleaner.api.Validate;
+import org.datacleaner.components.categories.WriteSuperCategory;
+import org.datacleaner.components.convert.ConvertToBooleanTransformer;
+import org.datacleaner.components.convert.ConvertToNumberTransformer;
+import org.datacleaner.connection.CsvDatastore;
+import org.datacleaner.connection.FileDatastore;
+import org.datacleaner.connection.SchemaNavigator;
+import org.datacleaner.connection.UpdateableDatastore;
+import org.datacleaner.connection.UpdateableDatastoreConnection;
+import org.datacleaner.data.MetaModelInputColumn;
+import org.datacleaner.descriptors.FilterDescriptor;
+import org.datacleaner.descriptors.TransformerDescriptor;
+import org.datacleaner.desktop.api.PrecedingComponentConsumer;
+import org.datacleaner.job.builder.AnalysisJobBuilder;
+import org.datacleaner.util.WriteBuffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Named("Update DataHub")
+@Description("Update records in a table in a registered datastore. This component allows you to map the values available in the flow with the columns of the target table, in order to update the values of these columns in the datastore."
+        + "\nTo understand the configuration of the Update table component, consider a typical SQL update statement:"
+        + "\n<blockquote>UPDATE table SET name = 'John Doe' WHERE id = 42</blockquote>"
+        + "\nHere we see that there is a condition (WHERE id=42) and a value to update (name should become 'John Doe'). This is what the two inputs are referring to. But obviously you are not dealing with constant values like 'John Doe' or '42'. You have a field in your DC job that you want to map to fields in your database."
+        + "\nUsually the 'condition value' would be a mapping of the key that you have in your job towards the key that is in the database. The 'values to update' property would include the columns that you wish to update based on the values you have in your job.")
+@Categorized(superCategory = WriteSuperCategory.class)
+@Concurrent(true)
+public class UpdateDataHubAnalyzer implements Analyzer<WriteDataResult>, Action<Iterable<Object[]>>, HasLabelAdvice,
```

Comment:
This was still the description copied from the update table component. A new, more specific, description has been added.